### ISSUE TITLE
Nullness Checker crash: AsSuperVisitor: type is not an erased subtype of supertype 11

### ISSUE THREAD

**Issue by wmdietlGC** [typetools/checker-framework#1420]
Thanks for the quick fix of #1419!
In a repo with master and #1387 (including fix for #1419 ) put the following into `Crash8Lib.java`:

```java
interface Crash8Lib {
  interface MyIterable<T> extends Iterable<T> {}

  interface MyComparable<K extends MyComparable<K>> extends Comparable<K> {}

  interface Box<E extends Box<E, K, C, I>,
                     K extends MyComparable<K>,
                       C extends Enum<C>, I extends Enum<I>> {
  }

  interface Root<T extends Root<T, R>, R> {
    T getT();
    MyIterable<R> getMIR();
  }

  interface Sub<T extends Sub<T, R, E>,
                  R, E extends Box<E, ?, ?, ?>>
      extends Root<T, R> {
  }

  interface Leaf<T extends Leaf<T, E>,
                   E extends Box<E, ?, ?, ?>>
      extends Sub<T, E, E> {
  }

  interface Main {
    <E extends Box<E, ?, ?, ?>>
       Leaf<?, E> foo(Class<E> b);
  }
}
```

and compile with or without the Nullness Checker.

Then put the following into `Crash8.java`:

```java
abstract class Crash8<E extends Crash8Lib.Box<E, ?, ?, ?>> {
  void test(Crash8Lib.Main main, boolean b, Class<E> cls) {
    Crash8Lib.MyIterable<E> x = null;
    x = b ? x
          : main.foo(cls).getT().getMIR();
  }
}
```

and compile with the Nullness Checker to get:

```
error: AsSuperVisitor: type is not an erased subtype of supertype.
  type: @Initialized @Nullable Object
  superType: Box<E extends Box<E, ? extends MyComparable<capture#42 of ? extends MyComparable<capture#42 of ? extends MyComparable<capture#42 of ?>>>, ? extends Enum<capture#43 of ? extends Enum<capture#43 of ? extends Enum<capture#43 of ?>>>, ? extends Enum<capture#44 of ? extends Enum<capture#44 of ? extends Enum<capture#44 of ?>>>>, ? extends MyComparable<capture#42 of ? extends MyComparable<capture#42 of ? extends MyComparable<capture#42 of ?>>>, ? extends Enum<capture#43 of ? extends Enum<capture#43 of ? extends Enum<capture#43 of ?>>>, ? extends Enum<capture#44 of ? extends Enum<capture#44 of ? extends Enum<capture#44 of ?>>>>
  Compilation unit: Crash8.java
  Exception: java.lang.Throwable; Stack trace: org.checkerframework.framework.source.SourceChecker.errorAbort(SourceChecker.java:727)
  org.checkerframework.javacutil.ErrorReporter.errorAbort(ErrorReporter.java:41)
  org.checkerframework.framework.type.AsSuperVisitor.errorTypeNotErasedSubtypeOfSuperType(AsSuperVisitor.java:155)
  org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:340)
  org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:29)
  org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:308)
  org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:59)
  org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:86)
  org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_Typevar(AsSuperVisitor.java:612)
  org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_Typevar(AsSuperVisitor.java:29)
  org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:662)
  org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:59)
  org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:86)
  org.checkerframework.framework.type.AsSuperVisitor.asSuper(AsSuperVisitor.java:68)
  org.checkerframework.framework.util.AnnotatedTypes.asSuper(AnnotatedTypes.java:131)
  org.checkerframework.framework.util.AtmLubVisitor.lubTypeArgument(AtmLubVisitor.java:198)
  org.checkerframework.framework.util.AtmLubVisitor.visitDeclared_Declared(AtmLubVisitor.java:184)
  org.checkerframework.framework.util.AtmLubVisitor.visitDeclared_Declared(AtmLubVisitor.java:31)
  org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:308)
  org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:59)
  org.checkerframework.framework.util.AtmLubVisitor.lub(AtmLubVisitor.java:66)
  org.checkerframework.framework.util.AnnotatedTypes.leastUpperBound(AnnotatedTypes.java:629)
  org.checkerframework.framework.type.TypeFromExpressionVisitor.visitConditionalExpression(TypeFromExpressionVisitor.java:105)
  org.checkerframework.framework.type.TypeFromExpressionVisitor.visitConditionalExpression(TypeFromExpressionVisitor.java:45)
  com.sun.tools.javac.tree.JCTree$JCConditional.accept(JCTree.java:1245)
  com.sun.source.util.SimpleTreeVisitor.visit(SimpleTreeVisitor.java:53)
  org.checkerframework.framework.type.TypeFromTree.fromExpression(TypeFromTree.java:36)
  org.checkerframework.framework.type.AnnotatedTypeFactory.fromExpression(AnnotatedTypeFactory.java:1196)
  org.checkerframework.framework.type.AnnotatedTypeFactory.getAnnotatedType(AnnotatedTypeFactory.java:970)
  org.checkerframework.framework.flow.CFAbstractTransfer.getValueFromFactory(CFAbstractTransfer.java:202)
  org.checkerframework.framework.flow.CFAbstractTransfer.visitNode(CFAbstractTransfer.java:549)
  org.checkerframework.framework.flow.CFAbstractTransfer.visitNode(CFAbstractTransfer.java:94)
  org.checkerframework.dataflow.cfg.node.AbstractNodeVisitor.visitTernaryExpression(AbstractNodeVisitor.java:217)
  org.checkerframework.framework.flow.CFAbstractTransfer.visitTernaryExpression(CFAbstractTransfer.java:649)
  org.checkerframework.framework.flow.CFAbstractTransfer.visitTernaryExpression(CFAbstractTransfer.java:94)
  org.checkerframework.dataflow.cfg.node.TernaryExpressionNode.accept(TernaryExpressionNode.java:56)
  org.checkerframework.dataflow.analysis.Analysis.callTransferFunction(Analysis.java:404)
  org.checkerframework.dataflow.analysis.Analysis.performAnalysis(Analysis.java:200)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1103)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1060)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.performFlowAnalysis(GenericAnnotatedTypeFactory.java:990)
```

I didn't manage to reproduce without the separate compilation and tried to minimize as much as possible. This is finally a reproduction for Crash8, which I've been trying for a while.

**Comment by smillst** [comment]
This is fixed in #1387.  (Note, this wasn't a problem with type argument inference, but rather an issue with the unbounded wildcard is `Leaf<?, E> foo(Class<E> b)`.

**Comment by smillst** [comment]
Fixed via https://github.com/typetools/checker-framework/commit/a03f6baf82af518b46c18f7f3765688b5767b821.
